apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -112,6 +118,14 @@ type ClusterQueueSpec struct {
 	StopPolicy *StopPolicy `json:"stopPolicy,omitempty"`
 }
 
+type AdmissionCheckStrategy struct {
```

Comment:
Do we support mutability of the structure? If not, I would suggest to validate it in the webhook.

Reply:
Interesting, so do we reflect the changes at the CQ level into the workload level (`status.admissionCheckStates`), or just let it continue as initialized?

Reply:
We propagate those changes into the workloads

Reply:
I see, makes sense, I'm yet wondering about some scenarios. For example, if an admin adds ProvReq to CQ with a running workload, is the workload evicted or continues?

Reply:
Yes, I've also wondered about scenarios like these. I imagine we wouldn't like to evict already running workload, similarly as it's done right now iiuc. However, introducing the new API also allows us to extend it, and maybe evicting (or not) could be also part of it in the future

Reply:
sgtm, I'm ok keeping it mutable since the current `admissionChecks` are mutable. I suppose we would like not to evict by default indeed, but we can handle this in the future if there is a need / user request.